### PR TITLE
feat: adding llama support via replicate and ollama

### PR DESCRIPTION
**Pull request by chr-hertel**

### CODE REVIEW

File: examples/chat-llama-replicate.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+use PhpLlm\LlmChain\Message\Message;
+use PhpLlm\LlmChain\Message\MessageBag;
+use PhpLlm\LlmChain\Model\Language\Llama;
+use PhpLlm\LlmChain\Platform\Replicate;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\HttpClient\HttpClient;
+
+require_once dirname(__DIR__).'/vendor/autoload.php';
+(new Dotenv())->loadEnv(dirname(__DIR__).'/.env');
+
```
Comment (OskarStark) [inline examples/chat-llama-replicate.php:13]: Please check for the env var

File: src/Platform/Replicate.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            sleep(1);
+
+            $response = $this->getResponse($response['id']);
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param string $id
```
Comment (OskarStark) [inline src/Platform/Replicate.php:43]: Can be removed

File: src/Model/Language/Llama.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+use PhpLlm\LlmChain\Message\MessageBag;
+use PhpLlm\LlmChain\Platform\Ollama;
+use PhpLlm\LlmChain\Platform\Replicate;
+use PhpLlm\LlmChain\Response\Choice;
+use PhpLlm\LlmChain\Response\Response;
+
+final readonly class Llama implements LanguageModel
+{
+    public function __construct(
+        private Replicate|Ollama $platform,
```
Comment (chr-hertel) [inline src/Model/Language/Llama.php:17]: ugly - better to have an Llama adapter for both

File: src/Model/Language/Llama.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public function call(MessageBag $messages, array $options = []): Response
+    {
+        $systemMessage = $messages->getSystemMessage();
+        $endpoint = $this->platform instanceof Replicate ? 'predictions' : 'chat';
+
+        $response = $this->platform->request('meta/meta-llama-3.1-405b-instruct', $endpoint, [
+            'system' => $systemMessage?->content,
+            'prompt' => $messages->withoutSystemMessage()->getIterator()->current()->content[0]->text, // @phpstan-ignore-line TODO: Multiple messages
```
Comment (chr-hertel) [inline src/Model/Language/Llama.php:28]: todo: properly convert `MessageBag` into Llama prompt

File: src/Platform/Ollama.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        $url = sprintf('%s/api/%s', $this->hostUrl, $endpoint);
+
+        $response = $this->httpClient->request('POST', $url, [
+            'headers' => ['Content-Type' => 'application/json'],
+            'json' => array_merge($body, [
+                'model' => $model,
+            ]),
+        ]);
+
+        return dump($response->toArray());
```
Comment (chr-hertel) [inline src/Platform/Ollama.php:34]: TODO: no words needed

File: src/Platform/Replicate.php [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace PhpLlm\LlmChain\Platform;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final readonly class Replicate
+{
+    public function __construct(
+        private HttpClientInterface $httpClient,
+        private string $apiKey,
```
Comment (OskarStark) [inline src/Platform/Replicate.php:13]:
```suggestion
        #[\SensitiveParameter]
        private string $apiKey,
```

**Pull request merged**